orderbook/src/database/trades.rs

```diff
@@ -0,0 +1,378 @@
+            .block_number
+            .try_into()
+            .context("block_number is not u32")?;
+        let log_index = self.log_index.try_into().context("log_index is not u32")?;
+        let order_uid = OrderUid(
+            self.order_uid
+                .try_into()
+                .map_err(|_| anyhow!("order uid has wrong length"))?,
+        );
+        let buy_amount = big_decimal_to_big_uint(&self.buy_amount)
+            .ok_or_else(|| anyhow!("buy_amount is not an unsigned integer"))?;
+        let sell_amount = big_decimal_to_big_uint(&self.sell_amount)
+            .ok_or_else(|| anyhow!("sell_amount is not an unsigned integer"))?;
+        let sell_amount_before_fees = big_decimal_to_big_uint(&self.sell_amount_before_fees)
+            .ok_or_else(|| anyhow!("sell_amount_before_fees is not an unsigned integer"))?;
+        let owner = h160_from_vec(self.owner)?;
+        let buy_token = h160_from_vec(self.buy_token)?;
+        let sell_token = h160_from_vec(self.sell_token)?;
+        Ok(Trade {
+            block_number,
+            log_index,
+            order_uid,
+            buy_amount,
+            sell_amount,
+            sell_amount_before_fees,
+            owner,
+            buy_token,
+            sell_token,
+        })
+    }
+}
+
+#[cfg(test)]
+mod tests {
+
+    use super::*;
+    use crate::database::{DbTrade, Event, EventIndex};
+    use ethcontract::U256;
+    use futures::StreamExt;
+    use model::order::{Order, OrderCreation, OrderMetaData};
+    use model::trade::Trade;
+    use num_bigint::BigUint;
+    use std::collections::HashSet;
+
+    #[tokio::test]
+    #[ignore]
+    async fn postgres_trade_roundtrip() {
+        let db = Database::new("postgresql://").unwrap();
+        db.clear().await.unwrap();
+        let filter = TradeFilter::default();
+        assert!(db.trades(&filter).boxed().next().await.is_none());
+
+        // Common fields
+        let sell_token = H160::from_low_u64_be(1);
+        let buy_token = H160::from_low_u64_be(2);
+        let owner = H160::from_low_u64_be(3);
+        let executed_buy_amount = 7u32;
+        let executed_sell_amount = 3u32;
+        let fee_amount = 2u32;
+        let uid = OrderUid([1u8; 56]);
+
+        let trade = Trade {
+            block_number: 2,
+            log_index: 0,
+            order_uid: uid,
+            sell_amount: BigUint::from(executed_sell_amount),
+            buy_amount: BigUint::from(executed_buy_amount),
+            sell_amount_before_fees: BigUint::from(executed_sell_amount - fee_amount),
+            owner,
+            buy_token,
+            sell_token,
+        };
+        let order = Order {
+            order_meta_data: OrderMetaData {
+                owner,
+                uid,
+                executed_sell_amount_before_fees: BigUint::from(executed_sell_amount),
+                executed_fee_amount: BigUint::from(fee_amount),
+                executed_buy_amount: BigUint::from(executed_buy_amount),
```

Comment:
Don't think you need to set these because they are not part of the information that is stored in the db.